test/api/resources/accounts.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import APIResource from 'api/apiResource';
+import AccountResource from 'api/resources/accounts';
+
+describe('AccountResouce', () => {
+	const defaultBasePath = 'http://localhost:1234';
+	const defaultPath = '/accounts';
+
+	let LiskAPI;
+	let resource;
+
+	beforeEach(() => {
+		LiskAPI = {
+			headers: {},
+			fullURL: defaultBasePath,
+			hasAvailableNodes: () => {},
+			randomizeNodes: () => {},
+			banActiveNodeAndSelect: () => {},
+		};
+		resource = new AccountResource(LiskAPI);
+	});
+
+	describe('#constructor', () => {
+		it('should throw error without LiskAPI input', () => {
+			return (() => new AccountResource()).should.throw(
+				'Require LiskAPI instance to be initialized.',
+			);
+		});
+
+		it('should be instance of APIResource', () => {
+			return resource.should.be.instanceOf(APIResource);
+		});
+
+		it('should have correcrt full path', () => {
+			return resource.resourcePath.should.eql(
+				`${defaultBasePath}/api${defaultPath}`,
+			);
+		});
+
+		it('should set resource path', () => {
+			return resource.path.should.equal(defaultPath);
+		});
+
+		it('should have methods', () => {
+			return resource.should.have.keys('path', 'get');
```

Comment:
We already tested `path`.

Reply:
It is true, but once we move to the `chai` test, we will change this to `all.keys`.
Therefore, we should try to list all of them

Reply:
Not sure what difference `all` makes: that means that of the keys you list it should have all of them, not that the keys you list should be all the keys it has. At the very least the description of this test would have to change if it wasn't specifically about methods.